autoplay/data_tools/last_fm_data.py

```diff
@@ -21,24 +21,68 @@ def get_secrets():
     return api_key, secret
 
 
-def get_scrobbles(username: str):
-    """Get the scrobbles for a given user.
-
-    Args:
-        username (str): LastFM username
-    """
+def create_network():
+    """Create a pylast network for accessing the LastFM API"""
     api_key, secret = get_secrets()
     network = pl.LastFMNetwork(
         api_key=api_key,
         api_secret=secret
     )
+    return network
 
+
+def get_scrobbles(username: str, limit: int = None):
+    """Get the scrobbles for a given user.
+
+    Args:
+        username (str): LastFM username
+        limit (int): the number of scrobbles to get. Defaults to None.
+    """
+    network = create_network()
     user = network.get_user(username)
-    scrobbles = user.get_recent_tracks(limit=None)
+    scrobbles = user.get_recent_tracks(limit=limit)
 
     return scrobbles
 
 
+def get_top_tags(scrobble: NamedTuple, tags_kept: int = 15):
+    """Get up to tags_kept tags for a track if possible
+
+    Args:
+        scrobble (NamedTuple): LastFM (pylast) class of song (PlayedTrack)
+        tags_kept (int): The number of top tags to keep
+
+    Returns:
+        a list of dicts: the tag (key) and its weight (value)
+    """
+    track = scrobble.track
+    top_tags = []
+    top_tags.extend(track.get_top_tags(limit=tags_kept))
+    """Slows down scraping significantly, commented out until a faster method is found"""
+    # Add more tags from album and artist if not enough
+    # if len(top_tags) < tags_kept:
+    #     try:
+    #         top_tags.extend(track.get_album().get_top_tags(limit = tags_kept - len(top_tags)))
+    #     # cannot find album via track, search for it
+    #     except (AttributeError, pl.WSError) as e:
+    #         network = create_network()
+    #         try:
+    #             network.search_for_album(scrobble.album).get_next_page()[0].get_top_tags(limit = tags_kept - len(top_tags))
+    #         # no results found for search
+    #         except IndexError:
+    #             print(scrobble.album)
+    #             pass
+    # if len(top_tags) < tags_kept:
+    #     top_tags.extend(track.get_artist().get_top_tags(limit = tags_kept - len(top_tags)))
+
```

Comment:
Shouldnt have this in code really, we could discuss but I am of the opinion that we should not worry about speed at first but quality. If getting good data takes time then I have no issue refactoring or trying to speed up later